src/pages/EntityDetails/Model/Logs/ActionLogs/ActionLogs.tsx

```diff
@@ -281,15 +284,17 @@ export default function ActionLogs() {
         const hasSterr =
           actionFullDetails.status === "failed" && !!actionFullDetails.message;
         const stdout = hasStdout
-          ? actionFullDetails.log.map((m, i) => (
+          ? actionFullDetails.log?.map((m, i) => (
```

Comment:
Just a small observation. TypeScript will state that the type of `stdout` is `JSX.Element[] | undefined`, while I think that, from the definition of `hasStdout`, it can't be `undefined`. Was thinking if adding the redundant `&& actionFullDetails.log` on line 286 might make sense?

Reply:
I've improved this by assigning `actionFullDetails.log` to a variable and then using that variable when creating `hasStdout` and also inside the ternary. This way typescript knows that when `hasStdout` is `true` then the variable must not be `undefined`.

```typescript
const log = actionFullDetails.log;
const hasStdout = log && log.length > 0;
...
const stdout = hasStdout
    ? log.map((m, i) => (
```